### ISSUE TITLE
Passing Enum Type as Parameter to Query - Quarkus

### ISSUE THREAD

**Issue by MillerRidgeway** [smallrye/smallrye-graphql#287]
Hi all,

I am having issues passing an enum type as a parameter to a query call. I have the following relevant snippets:

**Enum Definition**
```
    @Enum("ConfigurationEnum")
    @Description("The available configuration names")
    public enum ConfigurationEnum{
        allocation_projection("allocation.projection.maxHistoryDays"),
        testing_thingy("testing.graphql.config.service.save");

        private final String configurationValue;
        private ConfigurationEnum(String configurationValue){
            this.configurationValue = configurationValue;
        }

        @Override
        public String toString() {
            return this.configurationValue;
        }
    }
```

**Resource endpoint**
```
    @Query("configurationByName")
    @Description("Get a configuration by name")
    @Transactional
    public Configuration getByName(@Name("name") ConfigurationEnum name){
        return Configuration.getByName(name.toString());
    }
```

**Functional Implementation**
```
    @ApplicationScoped
    @Entity
    @Table(name="CONFIGURATION")
    public class Configuration extends PanacheEntityBase {

    @Id
    @NotNull
    private String name;
    private String value;

    public static Configuration getByName(String name){
        return find("name", name).firstResult();
    }
```

**My example query:**

```
query searchByName {
  configurationByName(name: allocation_projection) {
    name
    value
  }
}
```

**The resulting error:**

`"message": "Validation error of type WrongType: argument 'name' with value 'StringValue{value='allocation_projection'}' is not a valid 'Unknown Scalar Type `

<hr>

The desired behavior is to pass in the argument to `configurationByName` as an _enum_, then update the corresponding entry in the database field - the enum value. Using a scalar value - i.e. passing the string directly - everything works as expected. I need to change the type of the argument from a scalar to an enum. Generally, I believe this is done via setting the field type inside of `buildQueryType` when constructing custom schema. Is there any way to set this option in MicroProfile?

I'm pretty new to GraphQL in general and especially to the MicroProfile paradigm, so please forgive any simple oversights I may have made - I can't find any documentation with a similar case using MicroProfile.

Thanks!

**Comment by phillip-kruger** [comment]
Thanks for this !

**Comment by phillip-kruger** [comment]
Hi @MillerRidgeway - Thanks for this. I have fixed it and added test cases for this. This fix will be part of version 1.0.4 that I hope to get into Quarkus 1.6.
